index.rst

```diff
@@ -159,11 +159,43 @@ Pre-emption still has to be understood in terms of the registry sync jobs and re
 This approach allows the submission system to decide whether the registry is updated multiple times within the graph or solely at the end since the registry merging jobs can be configured to either merge the inputs and pass them on complete, or merge, sync and trim.
 A trimmed registry can not be passed on to the next job if the remainder has not been synced with the cloud registry.
 
+Limited Read-Only Registry
+--------------------------
+
+When a quantum graph is constructed the graph builder knows every single input dataset and every output dataset and how they relate to each other.
+The graph builder also knows the expected URIs of all the files created by the pipelines.
+This knowledge could be used to construct a limited in-memory non-SQL registry that could be constructed by reading a static file created by the graph builder.
```

Comment:
This is certainly an option, but I think "a small prepopulated SQLite registry" can _also_ do everything proposed in this section. It's overkill in a lot of ways, but it can also do pretty much everything we need already.

I would be inclined to go with the the SQLite version first, as the easiest thing to stand up, and then look for alternatives later (and ideally after with think about how we propagate provenance, which may make continuing to use SQLite more attractive).

Reply:
Funnily enough this morning as I finished the rest of the text I had convinced myself that I was using a trivial sqlite registry but clearly hadn't written that when I started writing a couple of days ago. Then this limited registry would be a subclass of the normal SQL registry but which behaves differently for writes.

Reply:
Another option might be to leave `Registry` and `Butler` as they are, and instead make a variant of `pipe.base.ButlerQuantumContext` that does this, just because that's a thinner interface that's already well-defined.  I think I'd try that first.  A simple `Butler` that uses the same `Registry` - but different, e.g. `getDataset` instead of `insertDatasets` on `put` - is another option.